packages/server-functions-plugin/src/index.ts

```diff
@@ -262,6 +272,24 @@ function resolveViteId(id: string) {
   return `\0${id}`
 }
 
+function buildFunctionId(opts: {
+  functionId: FunctionIdFn
+  directiveFnsById: Record<string, DirectiveFn>
+}): FunctionIdFn {
+  return (functionIdOps) => {
+    let generatedId = opts.functionId(functionIdOps)
+    if (generatedId in opts.directiveFnsById) {
+      let deduplicatedId = generatedId
+      let iteration = 0
+      do {
+        deduplicatedId = `${deduplicatedId}_${++iteration}`
```

Comment:
i am wondering whether we should call the function id generator function a single time at the end of the build with all of the functions?

then the function id generator function is responsible to provide unique function ids.

this ensures that we dont modfiy any of the function ids supplied by the user.

Reply:
We need the functionId at generation time for the replacers. We cant defer this up to the end